File: convertToBase7.kt
class Solution {
    fun convertToBase7(num: Int): String {
        var num = num
        if (num == 0) {
            return "0"
        }
        val negative = num < 0
        num = Math.abs(num)
        val digits = StringBuffer()
        while (num > 0) {
            digits.append(num % 7)
            num /= 7
        }
        if (negative) {
            digits.append('-')
        }
        return digits.reverse().toString()
    }
}

File: convertToBase7.scala
object Solution {
    def convertToBase7(num: Int): String = {
        var varn:Int = num
        if (varn == 0) return "0"
        val negative = varn < 0
        varn = Math.abs(varn)
        val digits = new StringBuffer
        while ( {
            varn > 0
        }) {
        digits.append(varn % 7)
        varn /= 7
        }
        if (negative) digits.append('-')
        digits.reverse.toString
    }
}
